dadi/lib/handlers/image.js

```diff
@@ -382,11 +384,18 @@ ImageHandler.prototype.convert = function (stream, imageInfo) {
                 (width / height) !== (imageInfo.width / imageInfo.height) &&
                 !self.storageHandler.notFound
               ) {
+                console.log({
+                  left: crops.x1,
+                  top: crops.y1,
+                  width: crops.x2 - crops.x1,
+                  height: crops.y2 - crops.y1
+                })
+
                 sharpImage.extract({
                   left: crops.x1,
                   top: crops.y1,
-                  width: crops.x1 + crops.x2,
-                  height: crops.y1 + crops.y2
+                  width: crops.x2 - crops.x1,
```

Comment:
That's correct. Lwip's [crop](https://github.com/EyalAr/lwip#crop) expects `x1`, `y1`, `x2` and `y2`, with `x2` being `x1 + width-of-the-crop` and `y2` being `y2 + height-of-the-crop`.

Sharp's [extract](http://sharp.dimens.io/en/stable/api-operation/#extract) expects `x1`, `y1`, `width` and `height`, so it infers `x2` and `y2` by adding the coordinates to the dimensions.

Does that make sense?